File: 1669-merge-in-between-linked-lists/1669-merge-in-between-linked-lists.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    ListNode* mergeInBetween(ListNode* list1, int a, int b, ListNode* list2) {
	ListNode* res = new ListNode(-1, list1);
	ListNode* dummy = res;

	int cnt = 0;
	ListNode *first, *second;
	while (cnt <= b) {
		if (cnt == a)
			first = dummy;
		if (cnt == b)
			second = dummy->next;
		dummy = dummy->next;
		cnt++;
	}
	first->next = list2;
	while (list2->next)
		list2 = list2->next;
	list2->next = second->next;

	return res->next;
    }

};

File: 1669-merge-in-between-linked-lists/1669-merge-in-between-linked-lists.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode mergeInBetween(ListNode list1, int a, int b, ListNode list2) {
        ListNode dummyHead = new ListNode(0, list1);
        ListNode first = null, last = null;
        while (list1 != null) {
            if (a == 1) {
                first = list1;
            } else if (b == 0) {
                last = list1.next;
                break;
            }
            a--; b--; list1 = list1.next;
        }
        first.next = list2;
        while (first.next != null) {
            first = first.next;
        }
        first.next = last;
        return dummyHead.next;
    }
}